core/repository/api/src/main/java/org/eclipse/rdf4j/repository/util/Repositories.java

```diff
@@ -142,9 +208,43 @@ public static void consumeSilent(Repository repository, Consumer<RepositoryConne
 	}
 
 	/**
-	 * Opens a {@link RepositoryConnection} to the given Repository, sends the connection to the given
-	 * {@link Function}, before either rolling back the transaction if it failed, or committing the
-	 * transaction if it was successful.
+	 * Opens a {@link RepositoryConnection} to the given Repository without opening a transaction, sends the
+	 * connection to the given {@link Function}.
+	 * 
+	 * @param <T>
+	 *        The type of the return value.
+	 * @param repository
+	 *        The {@link Repository} to open a connection to.
+	 * @param processFunction
+	 *        A {@link Function} that performs an action on the connection and returns a result.
+	 * @return The result of applying the function.
+	 * @throws RepositoryException
+	 *         If there was an exception dealing with the Repository.
+	 * @throws UnknownTransactionStateException
+	 *         If the transaction state was not properly recognised. (Optional specific exception)
+	 */
+	public static <T> T getNoTransaction(Repository repository,
```

Comment:
This is the core new method added, all of the other NoTransaction methods link back to here.